pkg/stream/event_stream.go

```diff
@@ -0,0 +1,98 @@
+package stream
+
+import (
+	"github.com/confluentinc/confluent-kafka-go/kafka"
+	"github.com/kelseyhightower/envconfig"
+)
+
+const EnableEventStreams = true
+const DisableEventStreams = false
+
+//go:generate mockgen -source=event_stream.go -package=stream -destination=mock_event_stream.go
+
+type KafkaProducer interface {
+	Produce(msg *kafka.Message, deliveryChan chan kafka.Event) error
+	Flush(timeoutMs int) int
+	Close()
+}
+
+type EventEnvelope struct {
+	Name    string      `json:"name"`
+	Payload interface{} `json:"payload"`
+}
+
+type EventStreamWriter interface {
+	Write(key, value []byte) error
+	Close()
+}
+
+type KafkaConfig struct {
+	BootstrapServer  string `envconfig:"KAFKA_BOOTSTRAP_SERVER" required:"true"`
+	ClientID         string `envconfig:"KAFKA_CLIENT_ID" default:""`
+	ClientSecret     string `envconfig:"KAFKA_CLIENT_SECRET" default:""`
+	Topic            string `envconfig:"KAFKA_EVENT_STREAM_TOPIC" required:"true"`
+	SecurityProtocol string `envconfig:"KAFKA_SECURITY_PROTOCOL" default:""`
+	SASLMechanism    string `envconfig:"KAFKA_SASL_MECHANISM" default:""`
+}
+
+type KafkaWriter struct {
+	kafkaProducer KafkaProducer
+	kafkaTopic    string
+}
+
+func getKafkaCfg(config *KafkaConfig) *kafka.ConfigMap {
+	cfg := &kafka.ConfigMap{
+		"bootstrap.servers": config.BootstrapServer,
+	}
+	if config.ClientID != "" {
+		_ = cfg.SetKey("sasl.username", config.ClientID)
+	}
+	if config.ClientSecret != "" {
+		_ = cfg.SetKey("sasl.password", config.ClientSecret)
+	}
+	if config.SecurityProtocol != "" {
+		_ = cfg.SetKey("security.protocol", config.SecurityProtocol)
+	}
+	if config.SASLMechanism != "" {
+		_ = cfg.SetKey("sasl.mechanism", config.SASLMechanism)
+	}
+	return cfg
+
+}
+
+func NewKafkaWriter() (*KafkaWriter, error) {
+	config := &KafkaConfig{}
+	err := envconfig.Process("", config)
+	if err != nil {
+		return nil, err
+	}
+	cfg := getKafkaCfg(config)
```

Comment:
```suggestion
	configMap := getKafkaConfigMap(config)
```
It took me a while to understand why we need ``config`` and ``cfg`` :sweat_smile: